### DISCUSSION TITLE
"RangeError: Maximum call stack size exceeded" when building NextJS Projekt in a GitLab CI Pipeline

### DISCUSSION THREAD

**Discussion by ChristianRaoulis** [vercel/next.js#49851] [Help] [answered]
### Summary

I got a NextJS Projekt which get's build and deployed in a GitLab CI/CD Pipeline. The NextJS Project uses a [few dependencies](https://gist.github.com/Skyriis/018ba0d699c9d72cc0b9a3b19f2535e8) and the app dir.

For a few weeks now, however, I have had the problem that building the NextJS project has a high probability of failing (roughly 85%) and showing me an (for me) [incomprehensible error](https://gist.github.com/Skyriis/208c62e05fbf33cf4fef1d5d5d74e980).

As already noted, this error does not always occur. Even if I run the same pipeline multiple times, it happens that 1 in 10 attempts completes successfully. The build process is always successfully completed locally, which makes it extremely difficult to find the cause of the error.

**Comment by icyJoseph** [comment]
Hi,

Doesn't it look like a typescript error? I'd seem that trying to resolve a type runs into an infinite recursion.

It is not uncommon, albeit very rate anyway:

- https://github.com/microsoft/TypeScript/issues/53783, 
- https://github.com/microsoft/TypeScript/issues/53593

Are you perhaps using Styled Components? Or doing any heavy TypeScript?

**Reply by icyJoseph** [reply, marked as answer]
I found one such `maximum call stack size exceeded` error that matches your error dump: https://github.com/microsoft/TypeScript/issues/46119

**Reply by ChristianRaoulis** [reply]
> Are you perhaps using Styled Components? Or doing any heavy TypeScript?

I do use some styled components and i'm using `ts-auto-guard` to generate some type guards which are used to validate api response types in a custom redux toolkit query. I suppose those type guards are pretty heavy typescript.

Is there a way to workaround that issue?

Edit:

one of the biggest questions for me is why this error only occurs in gitlab runner. If I build the project locally then it does not occur and the process completes successfully. Any ideas?

**Reply by icyJoseph** [reply]
Perhaps some requests fail and triggers a failure branch? But if it is TS failing, then it must be a compile time error, before requests happen.

Or the process runs out of the memory? Maybe the recursion is not infinite, just very deep? How much RAM does the gitlab runner have?

And, yeah not sure what workaround we could use here 😓  it is one of those bugs that are hard to manifest.

I had some issue with some styled components long ago, and I fixed it by downgrading the styled components types library, that is `@types/styled-components` -

**Reply by ChristianRaoulis** [reply]
After some testing around, I'm now absolutely sure that the ts-auto-guard type guards are causing the problem. The generated if checks are simply too long. Thank you for helping me to find the cause :)

**Comment by vishnu-vv** [comment]
Updating the version of Typescript solved the issue for me! 

Definitely try that out
